zulip_bots/zulip_bots/bots/calendarbot/calendarbot.py

```diff
@@ -1,154 +1,250 @@
+        num_args = len(filtered_args)
+        duration = DEFAULT_EVENT_DURATION
+
+        # Validate args Zulip global time and duration
+        if not num_args or filtered_args[0] == "help":
+            return self.usage()
+        elif filtered_args[0] == "auth":
+            authenticate_google()
+            return "Authenticating Google token..."
+        elif num_args > 2: 
+            return f"Expected at most 2 arguments, received {num_args}"
+
+        if not search(TIME_FORMAT, filtered_args[0]):
+            return f"Could not parse '{message['content']}'. Please pass in a Zulip global time `<time`"
+
         try:
-            meeting_details = self.parse_meeting_details(datetime_input,length_minutes)
-            self.create_calendar_event(meeting_details,cal)
-            self.send_google_event_invite(meeting_details)
-            self.send_event_file(cal)
+            if num_args == 2:
+                duration = int(filtered_args[1])
+                bot_handler.storage.put("duration", duration)
         except:
-            logging.exception('Error occurred during parsing.')
-            self.input_error_reply()
+            return f"Could not parse duration input {filtered_args[1]}"
 
-    def send_google_event_invite(self,meeting_details):
-        try:
-            GcalMeeting(meeting_details).send_event()
-        except AuthenticationError: 
-            self.authentication_error_reply()
+        time = filtered_args[0].replace("<time:", "").replace(">", "")
+        bot_handler.storage.put("starttime", time)
 
-    def parse_meeting_details(self,datetime_input,length_minutes) -> MeetingDetails:
-        
-        #converts Zulip global timepicker into a python datetime object
-        meeting_start, meeting_end = self.parse_datetime_input(datetime_input, length_minutes)
-        
-        name = "test name"
-        summary = "test summary"
-        sender_id = self.message['sender_id']
-        
+        confirm_message = f"Create a meeting at {filtered_args[0]} for {duration} mins? (Y / N)"
+
+        return confirm_message
+    
+    
+    # TODO: Check what occurs when a different sender pings bot
+    def parse_second_message(self, message: Dict[str, Any], bot_handler: BotHandler, starttime: str, cal: Calendar, duration: Optional[str] = DEFAULT_EVENT_DURATION) -> None:
+        """
+        Process confirmation for meeting details in temporary bot_handler storage
+        """
+
+        filtered_args = self.message_content_helper(message)
+        confirmation = filtered_args[0]
+
+        if confirmation == "y":
+            duration = int(duration)
+            meeting_start, meeting_end = self.parse_datetime_input(starttime, duration)
+            meeting_details = self.create_meeting_details(meeting_start, meeting_end, message["sender_email"], message["display_recipient"], duration)
+
+            # Create event ics file for download 
+            self.create_ics_event(meeting_details, cal)
```

Comment:
This smells bad to me (i think I wrote it) but feels like a lot is happening here. Is it worth breaking out?

Reply:
Yes, definitely. I think it will be more obvs how to break out when writing tests. I need to go back to that...